x/nexus/types/types.go

```diff
@@ -55,9 +56,27 @@ func (m ChainState) Validate() error {
 		return err
 	}
 
+	if len(m.Assets) == 0 {
+		return fmt.Errorf("no assets found")
+	}
+
+	seenDenoms := make(map[string]bool)
+
 	for _, asset := range m.Assets {
-		if err := sdk.ValidateDenom(asset); err != nil {
-			return err
+		if err := utils.ValidateString(asset); err != nil {
```

Comment:
Are you sure `sdk.ValidateDenom` is a subset of `utils.ValidateString`?

Reply:
Yep, it is. We're a bit inconsistent about denom checks. Should I keep the original?

Reply:
Sorry, if `sdk.ValidateDenom` is a subset, it means it's actually stricter than `utils.ValidateString`. In that case, we should definitely use `sdk.ValidateDenom` instead of `utils.ValidateString` cuz otherwise cosmos may not like it.